src/core/protocol-authorization.ts

```diff
@@ -199,6 +199,43 @@ export class ProtocolAuthorization {
     }
   }
 
+  /**
+   * Verifies the `protocolPath` declared in the given message (if it is a RecordsWrite) matches the path of actual ancestor chain.
+   * @throws {DwnError} if fails verification.
+   */
+  private static verifyProtocolPath(
+    inboundMessage: RecordsRead | RecordsWrite,
+    ancestorMessageChain: RecordsWriteMessage[],
+    recordSchemaToLabelMap: Map<string, string>
+  ): void {
+    // skip verification if this is not a RecordsWrite
+    if (inboundMessage.message.descriptor.method !== DwnMethodName.Write) {
+      return;
+    }
+
+    const currentRecordSchema = inboundMessage.message.descriptor.schema!;
```

Comment:
Nit-ish: Not sure it's safe to say that `schema` is definitely defined. I just tested sending a RecordsWrite with undefined schema and was able to reach this line of code. `recordSchemaToLabelMap.get(currentRecordSchema)` handles undefined input without throwing, so this line runs, but if we want honest typescript, maybe we should explicitly check for undefined.

Reply:
Super keen eyes! Was hoping to slip this through under the radar. You are right, there was no enforcement for schema to be there for protocol-based records but there ought to be. I've updated the schema and tests to enforce so.

Was unsure if Daniel wants to allow a non-schema records to be able to written using DWN protocol and what the implication is. I am obviously not in favor of it simply because it invalidates a lot of our protocol authorization code due to its assumptions and it's likely a longer discussion than we want right now. So I will try to let this fly under the radar again by just go ahead and add the check for `schema` (notice I didn't actually tag Daniel). = ]

Reply:
> (notice I didn't actually tag Daniel). = ]

absolute anarchy. corporate espionage in action. i'd have it no other way